core/src/main/java/io/micronaut/core/annotation/AnnotationMetadata.java

```diff
@@ -390,7 +395,10 @@ default boolean isDeclaredAnnotationPresent(@NonNull Class<? extends Annotation>
      * @return The annotation name
      */
     default Optional<String> getAnnotationNameByStereotype(@Nullable String stereotype) {
-        return getAnnotationNamesByStereotype(stereotype).stream().findFirst();
+        for (String s : getAnnotationNamesByStereotype(stereotype)) {
+            return Optional.of(s);
+        }
+        return Optional.empty();
```

Comment:
Is looping even beneficial here? You can just assign to a variable, `if (size > 0) return .get(0) else empty`

Reply:
Looping is a equivalent to: get iterator; if (it.hasNext) return it.next

Reply:
@jameskleeh `get(0)` is probably better because of not creating an iterator but then it looks a bit strange

Reply:
Having said that I don't know if it would even perform better since you have to check the size. Would be interested to know

Reply:
Iterator also needs to check the size at least once, `get` call should reduce the iterator allocation. I have modified a few more cases in the class.